test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
```

Comment:
I tried sinon spy but they didn't offer that much value for such a simple case and I kept getting a strange error. Basically I gave up because I felt I was wasting time.